UnityProject/Assets/Scripts/Messages/Server/ServerMessage.cs

```diff
@@ -81,33 +81,9 @@ public static void SendTo(NetworkConnection recipient, T msg, int channel = 0)
 		/// Sends the network message only to players who are visible from the
 		/// worldPosition
 		/// </summary>
-		public static void SendToVisiblePlayers(Vector2 worldPosition, T msg, int channel = 0, bool DoLinecast = true)
+		public static void SendToVisiblePlayers(Vector2 worldPosition, T msg, int channel = 0, bool doLinecast = true)
 		{
-			//Player script is not null for these players
-			var players = PlayerList.Instance.InGamePlayers;
-
-			LayerMask layerMask = LayerMask.GetMask( "Door Closed");
-			for (int i = players.Count - 1; i > 0; i--)
-			{
-				if (Vector2.Distance(worldPosition,
-					players[i].Script.PlayerChatLocation.AssumedWorldPosServer()) > 14f)
-				{
-					//Player in the list is too far away for this message, remove them:
-					players.Remove(players[i]);
-					continue;
-				}
-
-				if (DoLinecast == false) continue;
-				//within range, but check if they are in another room or hiding behind a wall
-				if (MatrixManager.Linecast(worldPosition, LayerTypeSelection.Walls, layerMask,
-					    players[i].Script.PlayerChatLocation.AssumedWorldPosServer()).ItHit)
-				{
-					//if it hit a wall remove that player
-					players.Remove(players[i]);
-				}
-
-
-			}
+			var players = RandomUtils.GetVisiblePlayers(worldPosition, doLinecast);
```

Comment:
wait, why is that under ``RandomUtils``?

Reply:
not too sure where else to put it, should i make a new class?

Reply:
we don't have any other Utils class to dump stuff into? RandomUtils was supposed  to be used for randomization